src/Chainweb/BlockHeaderDB/RestAPI/Server.hs

```diff
@@ -303,7 +301,7 @@ headerStreamHandler db = Tagged $ \req resp -> do
 
     g :: BlockHeader -> IO HeaderUpdate
     g bh = do
-        x <- casLookupM cas $ _blockPayloadHash bh
+        Just x <- lookupPayloadWithHeight cas (_blockHeight bh) $ _blockPayloadHash bh
```

Comment:
If this partial match fails, we'll just abort. I'd like to see better error reporting here.